llvm/lib/Target/ARM/ARMBaseInstrInfo.cpp

```diff
@@ -6623,10 +6626,8 @@ void ARMBaseInstrInfo::buildOutlinedFrame(
       MBB.addLiveIn(ARM::LR);
 
     // Insert a save before the outlined region
-    bool Auth = OF.Candidates.front()
-                    .getMF()
-                    ->getInfo<ARMFunctionInfo>()
-                    ->shouldSignReturnAddress(true);
+    bool Auth = MF.getInfo<ARMFunctionInfo>()
+                 ->shouldSignReturnAddress(true);
     saveLROnStack(MBB, It, true, Auth);
```

Comment:
This is a largely NFC change, but it makes the usage of saveLROnStack more consistent between `insertOutlinedCall` and `buildOutlinedFrame`. 

Both `insertOutlinedCall` and `buildOutlinedFrame` can query `MF` for `shouldSignReturnAddress`, instead of `buildOutlinedFrame` grabbing the front candidate.